sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -71,22 +73,22 @@ async def delete_repository(self, repository_name: str, **kwargs: Dict[str, Any]
                 :dedent: 8
                 :caption: Delete a repository from the `ContainerRegistryClient`
         """
-        result = await self._client.container_registry.delete_repository(repository_name, **kwargs)
-        return DeleteRepositoryResult._from_generated(result)  # pylint: disable=protected-access
+        try:
+            await self._client.container_registry.delete_repository(repository, **kwargs)
+        except ResourceNotFoundError as exc:
+            if "Not Found" in exc.reason:
```

Comment:
Checking `reason` is generally redundant - as it's just a string code associated with the status code. It's not something the service configures.

Reply:
Ahh, the swallowing 404s has been added to the swagger so the error returned is pretty vague. I can check more in depth but the raised error would look like "Not Found" to the user.

Reply:
I added a callback to do a check to make sure "<object> UNKNOWN" was in the response body

Reply:
Yeah but there's no point in adding this. If there's no deserialized body to the error that we can check for details, then your only choice is to catch and pass `ResourceNotFoundError` in its entirety.